hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -2270,7 +2270,10 @@ public String getContainingTableExpression() {
 
 	@Override
 	public int getJdbcTypeCount() {
-		return foreignKeyDescriptor.getJdbcTypeCount();
+		if ( sideNature == ForeignKeyDescriptor.Nature.KEY ) {
+			return foreignKeyDescriptor.getJdbcTypeCount();
+		}
+		return 0;
```

Comment:
Not sure changing this is correct, but the difference in behavior is due to the fact that before we were using `forEachSelectable` that only visited selectables for the key side. This is to avoid adding a capturing lambda back, and making this method consistent with `getSelectable`.